### DISCUSSION TITLE
WebUI could not connect to Ollama

### DISCUSSION THREAD

**Discussion by guwan** [open-webui/open-webui#1518] [General]
# Bug Report
WebUI could not connect to Ollama

## Description
The open webui was unable to connect to Ollama, so I even uninstalled Docker and reinstalled it, but it didn't work. I have referred to the solution on the official website and tried it again, but it didn't work. https://docs.openwebui.com/getting-started/troubleshooting/

I also have this issue with Windows 11. When the webui is first started, it is normal, but after restarting the computer, it cannot connect to Ollama even when starting through Docker Desktop

**Bug Summary:**
WebUI could not connect to Ollama

**Steps to Reproduce:**
I have a newly installed server with the following configurations:

1. Ubuntu 23.10
2. Followed the official installation guide for Ollama, and installed the Gemma model.
3. Installed Docker using the command 
```
sudo apt-get install -y docker-ce docker-ce-cli containerd.io.
```
4. Run Open WebUI using the command line: 
```
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

**Expected Behavior:**
Connect Ollama normally in webui and select the model

**Actual Behavior:**
WebUI could not connect to Ollama

## Environment

1. Ubuntu 23
2. window11

## Reproduction Details

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I have included the browser console logs.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the Docker container logs.

## Logs and Screenshots
![image](https://github.com/open-webui/open-webui/assets/5222986/75bcb57a-eb63-4002-83eb-d4a7a089d265)

![image](https://github.com/open-webui/open-webui/assets/5222986/0777f1df-83f2-4f3d-b42c-6e454d2b012e)

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

```
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by guwan** [comment]
the following method was  used to solve the problem by running Docker:

```
docker run -d -p 3000:8080 --network=host --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```
ollama
```
http://127.0.0.1:11434
```


Then I accessed the webui service through serverIp: 8080

**Reply by superYuezhang** [reply]
> http://127.0.0.1:11434

Thanks！This is useful to me！

**Reply by reminx** [reply]
I have the same issue.  It is useful!

**Comment by nikolateslaz** [comment]
I believe this is just a workaround, and this approach causes the container to lose the network isolation features provided by Docker
